x-pack/plugins/observability_solution/synthetics/server/synthetics_service/synthetics_monitor/synthetics_monitor_client.ts

```diff
@@ -309,14 +307,20 @@ export class SyntheticsMonitorClient {
 
   async getAllMonitorConfigs({
     spaceId,
+    soClient,
     encryptedSavedObjects,
   }: {
     spaceId: string;
+    soClient: SavedObjectsClientContract;
     encryptedSavedObjects: EncryptedSavedObjectsPluginStart;
   }) {
     const paramsBySpacePromise = this.syntheticsService.getSyntheticsParams({ spaceId });
+    const monitorConfigRepository = new MonitorConfigRepository(
+      soClient,
+      encryptedSavedObjects.getClient()
```

Comment:
Isn't it possible to instantiate the SyntheticsMonitorClient with the MonitorConfigRepository directly? Maybe that's too much of a refactoring. If we were to move the SyntheticsMonitorClient into the SynthethicsRouterWrapper were we instantiate the repository, we could directly inject it as a dependency? 

Also, maybe not possible to do if the SyntheticsMonitorClient is used outside of the route, I haven't checked deeper